bentoml/adapters/annotated_image_input.py

```diff
@@ -0,0 +1,384 @@
+
+
+def read_json_file(json_file):
+    """
+    Read the provided JSON file and return the parsed Python object
+    json_file can be any text or binary file that supports .read()
+    """
+    try:
+        parsed = json.load(json_file)
+    except (json.JSONDecodeError, UnicodeDecodeError):
+        raise BadInput("BentoML#AnnotatedImageInput received invalid JSON file")
+
+    return parsed
+
+
+def verify_image_format_or_raise(file_name: str, accept_format_list: [str]):
+    """
+    Raise error if file's extension is not in the provided accept_format_list
+    """
+    _, extension = os.path.splitext(file_name)
+    if extension.lower() not in accept_format_list:
+        raise BadInput(
+            "Input file not in supported format list: {}".format(accept_format_list)
+        )
+
+
+class AnnotatedImageInput(BaseInputAdapter):
+    """Transform incoming image data from http request, cli or lambda event into a
+    numpy array, while allowing an optional JSON file for image annotations (such
+    as object bounding boxes, class labels, etc.)
+
+    Transforms input image file into a numpy array, and loads JSON file as
+    a JSON serializable Python object, providing them to user-defined
+    API functions.
+
+    Args:
+        image_input_name (string): An acceptable input name for HTTP request
+            and function keyword argument. Default value is "image"
+        annotation_input_name (string): An acceptable input name for HTTP request
+            and function keyword argument. Default value is "annotations"
+        accept_image_formats (string[]):  A list of acceptable image formats.
+            Default value is loaded from bentoml config
+            'apiserver/default_image_input_accept_file_extensions', which is
+            set to ['.jpg', '.png', '.jpeg', '.tiff', '.webp', '.bmp'] by default.
+            List of all supported format can be found here:
+            https://imageio.readthedocs.io/en/stable/formats.html
+        pilmode (string): The pilmode to be used for reading image file into numpy
+            array. Default value is 'RGB'.  Find more information at:
+            https://imageio.readthedocs.io/en/stable/format_png-pil.html
+
+    Raises:
+        ImportError: imageio package is required to use AnnotatedImageInput
+
+    Example:
+
+        >>> from bentoml import BentoService, api, artifacts
+        >>> from bentoml.artifact import TensorflowArtifact
+        >>> from bentoml.adapters import AnnotatedImageInput
+        >>>
+        >>> CLASS_NAMES = ['cat', 'dog']
+        >>>
+        >>> @artifacts([TensorflowArtifact('classifer')])
+        >>> class PetClassification(BentoService):
+        >>>    @api(input=AnnotatedImageInput())
+        >>>    def predict(self, image, annotations):
+        >>>        cropped_pets = some_pet_finder(image, annotations)
+        >>>        results = self.artifacts.classifer.predict(cropped_pets)
+        >>>        return [CLASS_NAMES[r] for r in results]
```

Comment:
could you also add a client API request example for this example service here?  A curl command or python code that makes a sample HTTP request to this service would work.  here's an example: https://github.com/bentoml/BentoML/blob/v0.8.5/bentoml/adapters/json_input.py

Reply:
Absolutely, will do!

Reply:
Added a HTTP and HTML form section to the docstring (similar to MultiImageInput)